esphome/components/globals/globals_component.h

```diff
@@ -45,16 +45,24 @@ template<typename T> class RestoringGlobalsComponent : public Component {
   float get_setup_priority() const override { return setup_priority::HARDWARE; }
 
   void loop() override {
+    store_value();
+  }
+
+  void on_shutdown() override {
+    store_value();
+  }
+
+  void set_name_hash(uint32_t name_hash) { this->name_hash_ = name_hash; }
+
+ protected:
+  void store_value() {
```

Comment:
```suggestion
  void store_value_() {
```
Needed to match the naming conventions